eslint-local-rules.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+/**
+ * This file contains any locally defined ESLint rules. They are picked up by
+ * eslint-plugin-local-rules and exposed as 'local-rules/<rule-name>'.
+ * See packages/@n8n_io/eslint-config/ for details.
+ *
+ * These are looked up in a directory above node_modules/, which means that for an
+ * npm workspace it needs to be located at the very root.
+ */
+module.exports = {
+	/**
+	 * A rule to detect calls to JSON.parse() that are not wrapped inside try/catch blocks.
+	 *
+	 * Valid:
+	 * ```js
+	 * try { JSON.parse(foo) } catch(err) { baz() }
+	 * ```
+	 *
+	 * Invalid:
+	 * ```js
+	 * JSON.parse(foo)
+	 * ```
+	 *
+	 * The pattern where an object is cloned with JSON.parse(JSON.stringify()) is allowed:
+	 *
+	 * Valid:
+	 * ```js
+	 * JSON.parse(JSON.stringify(foo))
+	 * ```
+	 */
+	'require-catch-json-parse': {
+		meta: {
+			type: 'problem',
+			docs: {
+				description: 'Calls to JSON.parse() must be surrounded with a try/catch block.',
+				recommended: 'error',
+			},
+			schema: [],
+			messages: {
+				requireCatchJsonParse: 'Surround the JSON.parse() call with a try/catch block.',
+			},
+		},
+		defaultOptions: [],
+		create(context) {
+			return {
+				MemberExpression(node) {
+					if (node.object.name !== 'JSON' || node.property.name !== 'parse') {
```

Comment:
Any comments welcome on how the rule itself could be written better. I'm aware it would be a bit different written with TS types and ruleTester, but I'm hoping it's somewhat understandable...

Reply:
I wholeheartedly agree with having the rules as TS and tests! I too discovered how having the rule written in Typescript already helps in catching some potential problems.

I was considering a tradeoff between having a single TS rule in a separate module (with a lot of boilerplate), and having the rule in JS stored locally, and this was just faster to get started with. It would be neat to combine the two, just not sure how the transpilation should be set up for it to be easy to use as a local rule (having it externalized would make this a non-issue). I'll think about this for a bit, let me know if you have good ideas dropping right off your sleeve 🤞🏻 

Also, great point about minimizing upwards lookups! I shall refactor this.